x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BpeTokenReader.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.nlp.tokenizers;
+
+import java.util.LinkedList;
+import java.util.Optional;
+import java.util.Queue;
+
+import static java.lang.Character.isDigit;
+import static java.lang.Character.isLetter;
+import static java.lang.Character.isWhitespace;
+
+/**
+ * This is a char sequence tokenizer that satisfies the regex bpe regex for gpt-2
+ *      "'s|'t|'re|'ve|'m|'ll|'d| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+"
+ *
```

Comment:
Please add to the comment that in plain English this boils down to one of:
1. An ASCII apostrophe followed by one of a handful of specific letter sequences
2. Optional space followed by an unbroken sequence of letters
3. Optional space followed by an unbroken sequence of numbers
4. Optional space followed by an unbroken sequence of characters that are neither letters, numbers nor whitespace
5. Non-empty sequence of whitespace